### ISSUE TITLE
CentOS/RHEL7 Brave support

### ISSUE THREAD

**Issue by seonadh** [brave/brave-browser#4280]
## Description 
Unable to install Brave browser on CentOS/RHEL 7 following corresponding instructions on: https://brave-browser.readthedocs.io/en/latest/installing-brave.html#linux

## Steps to Reproduce
   1. Install the latest version of Centos7
   2. rpm --import https://brave-browser-rpm-release.s3.brave.com/brave-core.asc
   3. Add the brave-browser-release repo
   4. yum install brave-keyring brave-browser

## Actual result:
The above will error out with a dependency error:

Error: Package: brave-browser-0.63.48-1.x86_64 (brave-browser-release)
           Requires: libc.so.6(GLIBC_2.18)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

## Expected result:
The latest version of RHEL/CentOS is 7, this is built with glibc2.17, to support the lastest RHEL/Centos there should not be a glibc2.18 dependency.
RHEL 8 beta has glibc2.18, no release of Centos8 is available currently

## Reproduces how often: 
Everytime

## Brave version (brave://version info)
brave-browser-0.63.48-1.x86_64

**Comment by hugobuddel** [comment]
The last supported release version seems to be 0.62.51: https://github.com/brave/brave-browser/releases/tag/v0.62.51

You can install it manually. (Which I have to do anyway, since I don't have root access on this work machine. Well this is nice actually. I can't upgrade automatically, so had to do it manually like every other week. Now manually upgrading doesn't work either for me, so peace of mind!)

I'm using Scientific Linux 7.6 (which is based on Centos 7, which is based on RHEL 7). This release is from December 2018, so certainly not old.

**Comment by NsCDE** [comment]
Hello,

Is there  any plans to recompile latest stable release for CentOS/RHEL? Or at least provide source rpm to do it myself.

It is a bit misleading to have exact instructions on BRave's page which are not working because they are linking to Fedora 29+ version of the Brave rpm.

Thanks in advance ...

**Comment by jseidl** [comment]
We're working on fixing our official repository with the appropriate version. Our latest release that installs cleanly on CentOS/RHEL 7 is `v0.62.51`. 

You can download the RPM here: https://github.com/brave/brave-browser/releases/tag/v0.62.51

You will need to [enable user namespaces](https://superuser.com/questions/1294215/is-it-safe-to-enable-user-namespaces-in-centos-7-4-and-how-to-do-it):

`echo 10000 > /proc/sys/user/max_user_namespaces`

**Comment by truatpasteurdotfr** [comment]
https://brave-browser.readthedocs.io/en/latest/installing-brave.html#linux for CentOS is misleading, 
Following the instructions
```
CentOS/RHEL

sudo rpm --import https://brave-browser-rpm-release.s3.brave.com/brave-core.asc

cat << EOF | sudo tee /etc/yum.repos.d/brave-browser-release.repo
[brave-browser-release]
name=Brave Browser Release Channel repository
baseurl=https://brave-browser-rpm-release.s3.brave.com/x86_64/
enabled=1
EOF

sudo yum install brave-browser
```
yields:
```
Error: Package: brave-browser-0.66.99-1.x86_64 (brave-browser-release)
           Requires: libc.so.6(GLIBC_2.18)(64bit)
```
It is not clearly stated that only version v0.62.51 is the last supported version for glibc-2.17 as stated in the previous comment.

**Comment by Zoktog** [comment]
This problem is still relevant, is there any solution to this problem? Does someone do it at all?

**Comment by bsclifton** [comment]
@Zoktog I believe because of the limitations, we're going to drop support for Cent7

cc: @jseidl for comment

**Comment by Zoktog** [comment]
very sorry. I think in 2019 this is the wrong policy. New young browser to abandon one of the platforms is to show attitudes towards users. Today it is centos7, tomorrow it will be another platform.

**Comment by DeveloperMCD** [comment]
Isn't there some workaround? Can it be manually compiled or something? Or would that cause conflicts with other versions of Glibc floating around on the system.

**Comment by truatpasteurdotfr** [comment]
I am using a singularity container with fedora30 and brave inside (/usr/bin/brave-browser 
--disable-metrics --no-first-run --no-sandbox )

**Comment by jseidl** [comment]
@DeveloperMCD  `glibc` is a very sensitive library. Having more than one version installed can cause a lot of issues. On my [previous comment](https://github.com/brave/brave-browser/issues/4280#issuecomment-504164269) there's the link to download the RPM of the latest working version. 

Chromium bumped the required `glibc` version a while ago thus breaking compatibility with CentOS 7. It isn't something super straightforward to bump back. 

Individual package maintainers are responsible for fixing that `glibc` dependency that apparently is automatically added by the linker during build time. We're currently evaluating the amount of work needed to remove this dependency thus enabling distros as `CentOS 7` that are still on `glibc` 2.17.

We'll keep both this issue and https://github.com/brave/devops/issues/1261 updated when he have further information on this matter.

**Comment by hugobuddel** [comment]
> I am using a singularity container with fedora30 and brave inside (/usr/bin/brave-browser
> --disable-metrics --no-first-run --no-sandbox )

Going to try this @truatpasteurdotfr. I'm new to singularity, so this is a good reason to try it. Maybe for Dropbox too, which also doesn't support Centos7. Do you have a image that we can pull?

How well does it integrate with the rest of your system? E.g. can you save files outside the singularity container etc?

Another issue I have is not having local root access, so can't install Brave through yum etc. Running it in a container would solve that too, doesn't it?

FWIW, I think it's fine with Brave dropping support for Centos 7 etc. Centos 8 should be out soon, and Brave has to move quickly to become the nr. 1 browser.

**Comment by truatpasteurdotfr** [comment]
> Going to try this @truatpasteurdotfr. I'm new to singularity, so this is a good reason to try it. Maybe for Dropbox too, which also doesn't support Centos7. Do you have a image that we can pull?

I have uploaded at library://tru/default/f30-brave, you can use is as is if you want to share your $HOME inside the container 
`singularity run -B /run  library://tru/default/f30-brave`

> How well does it integrate with the rest of your system? E.g. can you save files outside the singularity container etc?

singularity allows you to choose what is accessible from within the container, by default you have full access to your $HOME from the host.

> Another issue I have is not having local root access, so can't install Brave through yum etc. Running it in a container would solve that too, doesn't it?

yes, but you need to be root to install singularity (from sources, or with yum from EPEL).

**Comment by jseidl** [comment]
Just to update the community on this topic, we're still investigating this issue. We created a full CentOS build environment to make sure it is linked to the correct GLIBC but we've encountered some steps on our build process that are failing on this environment.

Our team is working hard to solve it and we'll keep you posted as soon as we have any new updates.

**Comment by fmarier** [comment]
Note: Brave works fine on CentOS 8, which has just been released.

Simply follow the [official installation instructions](https://brave-browser.readthedocs.io/en/latest/installing-brave.html#linux).

**Comment by fmarier** [comment]
Given that CentOS 8 is now out and that we do not currently have the resources to do the work necessary to fix the glibc and the keysigning issues specific to CentOS/RHEL 7, we have decided not to support that version at this time and to instead allocate our time to the Linux distros which are currently working.

**Comment by NoX1De** [comment]
@fmarier @hugobuddel Someone should REALLY update these posts above and on the Brave forum to make an obvious note and stop recommending that people install the "latest release that installs cleanly on CentOS/RHEL 7" is a serious risk from a security perspective.  It is very sad that CentOS/RHEL 7 is no longer supported as it's utilization is far higher than CentOS/RHEL 8 and will continue to be for years...

This is even more relevant now with this latest Chromium 0-day vulnerabilty, Brave version 0.62.51 is obviously still vulnerable to this and whatever other security issues/vulnerabilities have been addressed/patched since version 0.62.51 was released.  Maybe stop recommending that people install insecure older versions.

**Comment by fmarier** [comment]
@NoX1De I've removed the links from this issue. Do you have a link to the forum post that needs to be updated?

**Comment by daphil19** [comment]
I was able to get Brave running on CentOS 7 by installing the community-maintained `snap` https://snapcraft.io/brave

Currently, that version is 1.1.23, which still isn't quite up-to-date, but is much newer than 0.62.51, and at least creates the potential for future updates for those on CentOS 7.

**Comment by st1905** [comment]
This is to inform people who are having hard time installing/running Brave on Centos 7.

It works fine. Latest nightly or stable versions, all of them work using this method.

https://www.youtube.com/watch?v=1iLmueoq1d0

https://pastebin.com/vqjnw7tH
